lib/datadog/appsec/component.rb

```diff
@@ -28,12 +28,28 @@ def create_processor
 
       def initialize(processor:)
         @processor = processor
+        @mutex = Mutex.new
+      end
+
+      def reconfigure(ruleset:)
+        @mutex.synchronize do
+          old = @processor
+          @processor = Processor.new(ruleset: ruleset)
+
+          old.finalize if processor && processor.ready?
```

Comment:
Should we handle the case that the new processor is not ready and re-assign the old processor as the main processor?

Reply:
That would make sense.

Reply:
Done.